src/Google.Api.Gax/ApiCallRetryExtensions.cs

```diff
@@ -7,89 +7,87 @@
 
 using Google.Protobuf;
 using Grpc.Core;
+using System;
 using System.Threading;
+using System.Threading.Tasks;
 
 namespace Google.Api.Gax
 {
     public static class ApiCallRetryExtensions
```

Comment:
Could do, although it'll make the code in ApiCall very slightly less attractive (in my opinion, that is).
I like the current "_asyncCall.WithRetry(clock, scheduler)" that this allows.
Putting it in ApiCall would require "WithRetry(_asyncCall, clock, scheduler)"
A small difference I know.
I also quite like it being in a separate file, as these retry methods are entirely separate from the other logic in ApiCall, although this could be done as a partial class.